packages/vue-visible-on-scroll/tests/index.spec.js

```diff
@@ -0,0 +1,187 @@
+import { mount, createLocalVue } from '@vue/test-utils';
+import sinon from 'sinon';
+
+import VisibleOnScrollDirective from '@/index.js';
+
+const component = {
+  template: '<div><p class="show" v-visible-on-scroll>Test</p></div>',
+  directives: {
+    'visible-on-scroll': VisibleOnScrollDirective
+  }
+};
+
+const localVue = createLocalVue();
+
+const factory = () => mount(component, {
+  localVue,
+  mocks: {
+    $route: {}
+  }
+});
+
+describe('VisibleOnScroll', () => {
+  beforeAll(() => {
+    // Scroll & related events only work client side
+    process.browser = true;
+  });
+
+  it('adds window eventListeners for scroll, orientationChange and hashChange', () => {
+    sinon.spy(window, 'addEventListener');
+    factory();
+
+    expect(window.addEventListener.calledWith('scroll', sinon.match.func)).toBe(true);
+    expect(window.addEventListener.calledWith('orientationchange', sinon.match.func)).toBe(true);
+    expect(window.addEventListener.calledWith('hashchange', sinon.match.func)).toBe(true);
+  });
+
+  it('sets the default transformation on the element', () => {
+    const wrapper = factory();
+    const visibleOnScrollElement = wrapper.findComponent('p');
+    expect(visibleOnScrollElement.attributes('style')).toBe('transform: translate3d(0, 0, 0)');
+  });
+
+  describe('handleScroll', () => {
+    describe('on desktop', () => {
+      beforeAll(() => {
+        sinon.stub(document.documentElement, 'clientWidth').value(1200);
+      });
+      afterAll(() => {
+        sinon.restore();
+      });
+
+      describe('when at the top (150px) of the screen', () => {
+        describe('when scrolling down', () => {
+          describe('when the element is visible already (should always be the case)', () => {
+            it('updates the scrollPosition', () => {
+              const wrapper = factory();
+              const visibleOnScrollElement = wrapper.findComponent('p');
+              visibleOnScrollElement.element.scrollPosition = 20;
+              visibleOnScrollElement.element.scrolledVisible = true;
+
+              window.scrollY = 140;
+              window.dispatchEvent(new CustomEvent('scroll', {}));
```

Comment:
Why are the CustomEvent instantiations called with an empty object as second argument throughout these tests?

Reply:
It was taken from an example that had some arguments in the custom event and then just copy pasted for all the specs, but it's not needed.
Has been removed.